pkg/virtual/workspaces/registry/rest.go

```diff
@@ -286,14 +299,22 @@ func (s *REST) authorizeForUser(ctx context.Context, orgClusterName logicalclust
 
 	}
 
+	impersonatedClusterClient, err := s.impersonatedkubeClusterClient(&kuser.DefaultInfo{
+		Name:   currentUser.GetName(),
+		Groups: []string{kuser.SystemPrivilegedGroup},
```

Comment:
what is this doing? Without a clear explanation this look frightening.

Reply:
This is part of Splitted PR https://github.com/kcp-dev/kcp/pull/1456

Reply:
Fixed in commit https://github.com/kcp-dev/kcp/pull/1456/commits/001ae2f79dcd0fa6e0610661143936c0bc82898e or PR #1456

Reply:
This will be fixed through the soft impersonation as discussed.